privateSBOMExchange/src/petra/lib/internals/producer.py

```diff
@@ -0,0 +1,76 @@
+import copy
+import requests
+import os
+from petra.lib.models import DecryptVisitor
+from petra.lib.util.config import Config
+from petra.lib.internals.generator import Generator
+from cryptography import x509
+import tempfile
+from cryptography.hazmat.primitives import serialization
+
+kms_conf = Config("./config/kms_and_attribute-namespace.conf")
+
+class Producer:
+    def __init__(self, artifact, policy):
+        self.kms_url = kms_conf.get_kms_service_url()
+        self.sw_artifact = artifact
+        self.policy = policy
+        self.plaintext_sbom_tree = None
+        self.redacted_sbom_tree = None
+        self.cpabe_sk = ""
+        self.decrypted_sbom_tree = None
+        self.signing_cert = None
+
+    def redact_sbom(self):
```

Comment:
I don't think we should reuse function names in our API. As this is already a function in the Generator, I suggest changing the name to something a bit less ambiguous.
```suggestion
    def request_redaction(self):
```